File: backend/app/schemas/news.py
from typing import Optional
from app.schemas.base import BaseSchema


class SentimentScore(BaseSchema):
    label: str  # positive, negative, neutral
    score: float


class NewsArticle(BaseSchema):
    title: str
    summary: str
    link: str
    source: str
    published: str
    published_dt: Optional[str] = None
    sentiment: Optional[SentimentScore] = None


class NewsResponse(BaseSchema):
    articles: list[NewsArticle]
    total: int
    source: str

File: frontend/src/types/news.ts
export interface SentimentScore {
  label: "positive" | "negative" | "neutral";
  score: number;
}

export interface NewsArticle {
  title: string;
  summary: string;
  link: string;
  source: string;
  published: string;
  publishedDt: string | null;
  sentiment?: SentimentScore | null;
}

export interface NewsResponse {
  articles: NewsArticle[];
  total: number;
  source: string;
}
